### PR TITLE
Added ApcuStore as Redis alternative

### PR DESCRIPTION
**Pull request by SpazzMarticus**
Since there was no ApcuStore provided, I created one.
(Tests are skipped, if apcu extension does not exist.)

### CODE REVIEW

**Comment by SpazzMarticus** [comment]
In order to run the new tests the apcu extension needs to be activated in Travis CI. I don't know how to do that.

**Comment by ankitpokhrel** [comment]
Hi @SpazzMarticus , Thank you for your contribution. I will go through it as soon as I have some time.

To enable `apcu` extension in travis you can follow [this documentation](https://docs.travis-ci.com/user/languages/php/#installing-additional-php-extensions). Something like this should work (not tested though).

```yml
before_install:
  - echo yes | pecl upgrade apcu
  - echo "extension=apcu.so" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
```

**Comment by SpazzMarticus** [comment]
Thanks for your help. I managed to enable it.

File: src/Cache/ApcuStore.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        return apcu_store($this->getActualCacheKey($key), $contents, $this->getTtl());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function delete(string $key): bool
+    {
+        return apcu_delete($this->getActualCacheKey($key));
```
Comment (ankitpokhrel) [inline src/Cache/ApcuStore.php]: Just to make sure that the return type is always boolean and make scrutinizer happy (probably), we can do:

```suggestion
        return true === apcu_delete($this->getActualCacheKey($key));
```

Comment (SpazzMarticus) [reply src/Cache/ApcuStore.php]: Should not be a problem since `$key` is a string, but I too love a happy scrutinizer. 😄

**Comment by SpazzMarticus** [comment]
Thanks for the review! I think I fixed all problems mentioned.

**Comment by SpazzMarticus** [comment]
I changed the `CacheFactory`, but can't resolve the last _change requested_. Maybe you can override this.

**Comment by ankitpokhrel** [comment]
Thanks!

**Pull request merged**